File: mysql_cli.py
import sys
import pandas as pd
import pymysql
import json

# MySQL bağlantı bilgileri
host = 'localhost'
db = 'orcatradebot'
user = 'root'
password = 'MySql!bot2021Tr'
charset = 'utf8mb4'

if len(sys.argv) < 2:
    print('Kullanım: python mysql_cli.py "SQL_SORGUSU"')
    sys.exit(1)

sql = sys.argv[1]

try:
    conn = pymysql.connect(
        host=host,
        user=user,
        password=password,
        database=db,
        charset=charset,
        cursorclass=pymysql.cursors.DictCursor
    )
    with conn.cursor() as cursor:
        cursor.execute(sql)
        rows = cursor.fetchall()
        if not rows:
            print('(Boş sonuç)')
        else:
            for row in rows:
                print(json.dumps(row, ensure_ascii=False))
    conn.close()
except Exception as e:
    print(f'Hata: {e}')
    sys.exit(2)

File: mysql_cli.php
<?php
// Kullanım: php mysql_cli.php "SELECT * FROM tablo LIMIT 1"

// MySQL bağlantı bilgileri
$host = 'localhost';
$db   = 'orcatradebot';
$user = 'root';
$pass = 'MySql!bot2021Tr';
$charset = 'utf8mb4';

if ($argc < 2) {
    echo "Kullanım: php mysql_cli.php \"SQL_SORGUSU\"\n";
    exit(1);
}

$sql = $argv[1];

$dsn = "mysql:host=$host;dbname=$db;charset=$charset";
$options = [
    PDO::ATTR_ERRMODE            => PDO::ERRMODE_EXCEPTION,
    PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
    PDO::ATTR_EMULATE_PREPARES   => false,
];

try {
    $pdo = new PDO($dsn, $user, $pass, $options);
    $stmt = $pdo->query($sql);
    $results = $stmt->fetchAll();
    if (empty($results)) {
        echo "(Boş sonuç)\n";
    } else {
        foreach ($results as $row) {
            echo json_encode($row, JSON_UNESCAPED_UNICODE) . "\n";
        }
    }
} catch (PDOException $e) {
    echo "Hata: " . $e->getMessage() . "\n";
    exit(2);
} 